lib/SILOptimizer/Mandatory/OSLogOptimization.cpp

```diff
@@ -942,22 +903,34 @@ static void replaceAllUsesAndFixLifetimes(SILValue foldedVal,
     return;
   }
 
-  // Here, originalVal is not owned. Hence, borrow form foldedVal and use the
-  // borrow in place of originalVal. Also, destroy foldedVal at the end of its
-  // lifetime.
+  // Here, originalVal is guaranteed. It must belong to a borrow scope that
+  // begins at a scope introducing instruction e.g. begin_borrow or load_borrow.
+  // The foldedVal should also have been inserted at that point. Therefore,
+  // create a borrow of foldedVal at the beginning of the scope and use the
+  // borrow in place of the originalVal. Also, end the borrow and destroy
+  // foldedVal at the end of the borrow scope.
   assert(originalVal.getOwnershipKind() == ValueOwnershipKind::Guaranteed);
 
-  SILBuilderWithScope builder(&*std::next(foldedInst->getIterator()));
-  BeginBorrowInst *borrow =
-      builder.createBeginBorrow(foldedInst->getLoc(), foldedVal);
+  BorrowScopeIntroducingValue originalScopeBegin =
+      getUniqueBorrowScopeIntroducingValue(originalVal);
```

Comment:
How can you be confident that there will always be a single borrow introducer? If you're just being optimistic, then maybe this should bail out instead of asserting. Otherwise please comment.

Reply:
I am just being optimistic and could not think of a reason why this could happen. I can bail out instead of asserting.